### PR TITLE
Properly handle APIError failures in Project.up

### LINKED ISSUE #3333: Compose trying to attach to dead container after error reported?
Just noticed that there is an error from the `compose up` command but still says that it's trying to attach.  What's it trying to attach to here? 

    docker-compose up --build
    Recreating poc_1
    
    ERROR: for poc  Container command './entrypoint.sh' not found or does not exist.
    Attaching to

Might be just erroneous output.

### PR DESCRIPTION
**Pull request by shin-**
Fixes #3333

### CODE REVIEW

File: compose/project.py [hunk 1]
```diff
@@ -390,13 +390,16 @@ def do(service):
...
-        results = parallel.parallel_execute(
+        results, errors = parallel.parallel_execute(
             services,
             do,
             operator.attrgetter('name'),
             None,
             get_deps
         )
+        if errors:
+            raise ProjectError('Errors encountered during up phase.')
```
Comment (dnephin) [inline compose/project.py]: Maybe something like "Encountered errors while bringing up the project." ?

**Comment by dnephin** [comment]
I think this is the right idea, just one suggestion on wording.

**Comment by shin-** [comment]
Thanks! Updated the error message.

**Comment by dnephin** [comment]
LGTM

File: compose/project.py [hunk 2]
```diff
@@ -390,13 +390,18 @@ def do(service):
...
             services,
             do,
             operator.attrgetter('name'),
             None,
             get_deps
         )
+        if errors:
+            raise ProjectError(
+                'Encountered errors while bringing up the project.'
+            )
```
Comment (kojiromike) [inline compose/project.py]: Unfortunately this doesn't really let us know which services errored out.

Comment (dnephin) [inline compose/project.py]: I don't think that was the intent of this change. There was a different PR for that issue.

Comment (shin-) [inline compose/project.py]: The detailed error with service information is reported during the `parallel_execute` call ( https://github.com/docker/compose/blob/master/compose/parallel.py#L57 )

**Pull request merged**